src/meta-data/airqo_api.py

```diff
@@ -33,21 +34,21 @@ def update_site_meta_data(self, site_details: dict):
     def __request(self, endpoint, params=None, body=None, method=None):
 
```

Comment:
thanks @OchiengPaul442 :

Your attempt at refactoring the __request function to include the token as a query parameter looks good. It effectively adds the token parameter to the params dictionary for both PUT and POST requests, or creates a new params dictionary with only the token if params is None. This approach ensures that the token is included in the URL for authentication.

Here are a few suggestions for further improvement:

1. Error Handling: You're using a handle_api_error function, which is a good practice. However, it's essential to ensure that this function is defined and handles errors appropriately. You might consider passing an instance of your class to the __request method so that it can access the handle_api_error method within the class.

2. Code Reusability: If you have multiple methods that make API requests, you can further improve code reusability by centralizing common functionality like error handling and token handling.

3. Request URL: Constructing the URL using string concatenation (e.g., "%s/%s" % (self.AIRQO_BASE_URL, endpoint)) is fine, but you could use Python's f-strings (formatted strings) for improved readability. For example: f"{self.AIRQO_BASE_URL}/{endpoint}".

Here's a slightly modified version of your code with these suggestions in mind:

```
def __request(self, endpoint, params=None, body=None, method=None):
    try:
        headers = {"Content-Type": "application/json"}
        url = f"{self.AIRQO_BASE_URL}/{endpoint}"
        
        if params is None:
            params = {}
        params['token'] = self.API_AUTH_TOKEN

        if method == "put":
            api_request = requests.put(
                url,
                params=params,
                headers=headers,
                data=json.dumps(body),
                verify=False,
            )
        elif method == "post":
            api_request = requests.post(
                url,
                params=params,
                headers=headers,
                data=json.dumps(body),
                verify=False,
            )
        else:
            self.handle_api_error("Invalid")
            return None

        print(f"Request Url : {api_request.request.url}")
        print(f"Request Body : {api_request.request.body}")

        if api_request.status_code == 200 or api_request.status_code == 201:
            print(f"Request Response : {api_request.json()}")
        else:
            self.handle_api_error(api_request)
            return None
    except Exception as ex:
        print(ex)
        traceback.print_exc()
        return None

```

This code above above centralises token handling and URL construction, making it more concise and maintaining good readability.

Please review suggestions and share concerns/suggestions/thoughts accordingly.

Reply:
Thanks for the suggestions @Baalmart

Reply:
In the new commit, I have taken into account the new optimized code presented and also refactored a section to utilize the request function from the request library to reduce the code written and also make it more flexible for other HTTP requests.

```
if method == "put":
            api_request = requests.put(
                url,
                params=params,
                headers=headers,
                data=json.dumps(body),
                verify=False,
            )
        elif method == "post":
            api_request = requests.post(
                url,
                params=params,
                headers=headers,
                data=json.dumps(body),
                verify=False,
            )
        else:
            self.handle_api_error("Invalid")
            return None
```

Refactored to

`api_request = requests.request(method, url, params=params, headers=headers, data=json.dumps(body), verify=False)`